packages/api/src/APIProvider.tsx

```diff
@@ -130,13 +130,16 @@ const APIProvider = ({ children, standalone = false }: PropsWithChildren<TAPIPro
     const [environment, setEnvironment] = useState(getEnvironment(activeLoginid));
     const standaloneDerivAPI = useRef(standalone ? initializeDerivAPI(() => setReconnect(true)) : null);
 
-    const switchEnvironment = (loginid: string | null | undefined) => {
-        if (!standalone) return;
-        const currentEnvironment = getEnvironment(loginid);
-        if (currentEnvironment !== 'custom' && currentEnvironment !== environment) {
-            setEnvironment(currentEnvironment);
-        }
-    };
+    const switchEnvironment = useCallback(
+        (loginid: string | null | undefined) => {
+            if (!standalone) return;
+            const currentEnvironment = getEnvironment(loginid);
+            if (currentEnvironment !== 'custom' && currentEnvironment !== environment) {
+                setEnvironment(currentEnvironment);
+            }
+        },
+        [environment, standalone]
+    );
 
```

Comment:
@yashim-deriv 
wrapped with useCallback to avoid `useAuthorize` reruns, because we are using `switchEnvironment` in `useAuthorize` hook

Reply:
so if I switch environment from A to B, then from B to C, and then from C to A, its not gonna work, right? bcecasue this function was already memoised for parameter A.

useCallback should only be used either to prevent rerenders, or to memoise values of pure functions.

it should not be used to memoise functions where the purpose of function is the side effect, as it might actually prevent the side effect (aka "setEnvironment" function) from happening when it actually should happen (basically, when its coming to memoisation of functions with side effects, there is plenty of pitfalls, so really that should be avoided)

Reply:
after a chat with @heorhi-deriv - it does actually make sense to use useCallback